programs/bpf_loader/src/syscalls.rs

```diff
@@ -2118,976 +2311,3596 @@ fn call<'a>(
+                            (*bignum_result).copy_from_slice(&big_number_bytes);
+                        }
+                        // Smaller than our buffer
+                        else {
+                            let mut index = 0;
+                            for byte in big_number_bytes.iter() {
+                                (*bignum_result)[index] = *byte;
+                                index += 1;
+                            }
+                            *out_size = index as u64;
+                        }
+                        *is_negative = bn_result.is_negative() as u64;
+                        *result = Ok(0)
+                    }
+                }
+                Err(_) => *result = Err(SyscallError::BigNumberAddError.into()),
+            }
+        }
+    }
+}
+/// Subtract BigNums
+struct SyscallBigNumMul<'a> {
+    cost: u64,
+    word_cost: u64,
+    word_div_cost: u64,
+    compute_meter: Rc<RefCell<dyn ComputeMeter>>,
+    loader_id: &'a Pubkey,
+}
+impl<'a> SyscallObject<BpfError> for SyscallBigNumMul<'a> {
+    fn call(
+        &mut self,
+        arguments_addr: u64,
+        arguments_count: u64,
+        out_result_addr: u64,
+        out_negative_addr: u64,
+        out_size_addr: u64,
+        memory_mapping: &MemoryMapping,
+        result: &mut Result<u64, EbpfError<BpfError>>,
+    ) {
+        let my_func = String::from("sol_bignum_mul");
+        if arguments_count != 3 {
+            *result = Err(SyscallError::BigNumberArgError(my_func, 3u64, arguments_count).into())
+        } else {
+            let mut buffer = Vec::<BigNum>::new();
+            let mut bytes = 0u64;
+            translate_bignum_args(
+                arguments_addr,
+                arguments_count,
+                self.loader_id,
+                &mut buffer,
+                &mut bytes,
+                &memory_mapping,
+                result,
+            );
+            // Compute costs
+            question_mark!(
+                self.compute_meter
+                    .consume(calc_bignum_cost!(self, bytes as f64)),
+                result
+            );
+            // Get out size and out vector
+            let out_size = question_mark!(
+                translate_type_mut::<u64>(memory_mapping, out_size_addr, self.loader_id, true),
+                result
+            );
+            let bignum_result = question_mark!(
+                translate_slice_mut::<u8>(
+                    memory_mapping,
+                    out_result_addr,
+                    *out_size,
+                    self.loader_id,
+                    true
+                ),
+                result
+            );
+            // Get negative flag pointer
+            let is_negative = question_mark!(
+                translate_type_mut::<u64>(memory_mapping, out_negative_addr, self.loader_id, true),
+                result
+            );
+            // Execute function
+            let mut bn_result = match BigNum::new() {
```

Comment:
Possible to use `question_mark!` here?

Reply:
It balks because of the BigNum::ErrorStack does not have `into` or `Try`.... but this is one of those error rework items I mentioned .... should be able to map them to an error Enum that conforms